magenta-lib/src/main/scala/magenta/tasks/AWS.scala

```diff
@@ -47,28 +43,23 @@ trait Lambda extends AWS {
   }
 }
 
-trait ASG extends AWS {
-  def elb: ELB = ELB
-
-  def client(implicit keyRing: KeyRing) = {
-    val client = new AmazonAutoScalingClient(provider(keyRing), clientConfiguration)
-    client.setEndpoint("autoscaling.eu-west-1.amazonaws.com")
-    client
-  }
+object ASG extends AWS {
+  def makeAsgClient(keyRing: KeyRing, region: Region): AmazonAutoScalingClient =
+    new AmazonAutoScalingClient(provider(keyRing), clientConfiguration).withRegion(awsRegion(region))
 
-  def desiredCapacity(name: String, capacity: Int)(implicit keyRing: KeyRing) =
+  def desiredCapacity(name: String, capacity: Int)(implicit client: AmazonAutoScalingClient) =
```

Comment:
As a vague rule of thumb I use implicit args when I'm making a library and I want it to be easy for a third party to call. I use real args internally because they are clearer, unless it's something where there's a clear benefit. In these cases I'm not sure that writing `(implicit client: AmazonAutoScalingClient)` everywhere is so much better than writing `, client)` everywhere instead. No big deal.